pkg/kubelet/prober/results/results_manager.go

```diff
@@ -87,3 +115,26 @@ func (m *manager) Remove(id kubecontainer.ContainerID) {
 	defer m.Unlock()
 	delete(m.cache, id)
 }
+
+func (m *manager) Updates() <-chan Update {
+	updates := m.updates.Load().(chan Update)
+	if updates == nil {
+		m.Lock()
+		defer m.Unlock()
+		// Verify updatse did not change while we were aquiring the lock
+		updates = m.updates.Load().(chan Update)
+		if updates == nil {
+			updates = make(chan Update)
```

Comment:
It's probably unnecessary, but I think a small buffer could probably smooth things out. I arbitrarily picked 20 - does that seem reasonable?